static/app/components/tours/tourContext.tsx

```diff
@@ -189,23 +183,54 @@ export function useTourReducer<T extends TourEnumType>(
     [orderedStepIds]
   );
 
+  const startTour = useCallback(
+    (stepId?: T) => {
+      dispatch({type: 'START_TOUR', stepId});
+      callbacks?.onStartTour?.(stepId);
+    },
+    [callbacks]
+  );
+
+  const endTour = useCallback(() => {
+    dispatch({type: 'END_TOUR'});
+    callbacks?.onEndTour?.();
+  }, [callbacks]);
+
+  const nextStep = useCallback(() => {
+    dispatch({type: 'NEXT_STEP'});
+  }, []);
```

Comment:
Looks good! Just FYI, if the `NEXT_STEP` action is called when there are no more steps, it will have the same result as the `END_TOUR` action. If that's the case, I'm wondering if we'd expect `callbacks.onEndTour()` to be invoked.